ui/src/environments/environmentLoader.ts

```diff
@@ -0,0 +1,20 @@
+import { environment as defaultEnvironment } from './environment';
+
+export const environmentLoader = new Promise<any>( (resolve) => {
+
+  let xmlhttp = new XMLHttpRequest(),
+    method = 'GET',
+    url = './assets/environments/environment.json';
+
+  xmlhttp.open(method, url, true);
+
+  xmlhttp.onload = function() {
+    if (xmlhttp.status === 200) {
+      resolve(JSON.parse(xmlhttp.responseText));
+    } else {
+      resolve(defaultEnvironment);
```

Comment:
This will lead to very strange behaviors. We should be showing an error in this case.

This could perhaps be a follow-up fix, but there should at least be a TODO here. I would prefer that we never attempt to continue running the application if this JSON load fails, as the behavior will be undefined.